src/sentry/issues/group_policy.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, Optional, Set, Type
+
+from django.conf import settings
+
+from sentry.issues.grouptype import GroupType
+from sentry.models import Organization, Project
+from sentry.utils import metrics, redis
+
+if TYPE_CHECKING:
+    from sentry.utils.performance_issues.performance_detection import PerformanceProblem
+
+_group_policy_registry: Dict[int, Type[GroupPolicy]] = {}
+DEFAULT_IGNORE_LIMIT: int = 3
+DEFAULT_EXPIRY_TIME: int = 60 * 60 * 24
+
+
+@dataclass(frozen=True)
+class NoiseConfig:
+    ignore_limit: int = DEFAULT_IGNORE_LIMIT
+    expiry_time: int = DEFAULT_EXPIRY_TIME
+
+
+@dataclass(frozen=True)
+class GroupPolicy:
+    group_type_id: int
+    organizations: Optional[Dict[str, NoiseConfig]] = None
+    early_access: Optional[NoiseConfig] = None
+    default: NoiseConfig = NoiseConfig()
+
+    def __init_subclass__(cls: Type[GroupPolicy], **kwargs: Any) -> None:
+        super().__init_subclass__(**kwargs)
+        if _group_policy_registry.get(cls.group_type_id):
+            raise ValueError(
+                f"A group policy for the group type {cls.group_type_id} has already been registered."
+            )
+        _group_policy_registry[cls.group_type_id] = cls
+
+    def __post_init__(self) -> None:
+        # ensure that ignore limits only increase with rollouts
+        prev_limit = 0
+        if self.organizations:
+            for organization, noise_config in self.organizations.items():
+                if noise_config.ignore_limit > prev_limit:
+                    prev_limit = noise_config.ignore_limit
+
+        if self.early_access:
+            if self.early_access.ignore_limit < prev_limit:
+                raise ValueError(
+                    "Early Access ignore limit must be lower than Limited Access organizations"
+                )
+            prev_limit = self.early_access.ignore_limit
+
+        if self.default.ignore_limit < prev_limit:
+            raise ValueError(
+                "Default ignore limit must be lower than Early Access and Limited Access organization limits"
+            )
+
+
+def get_noise_config(group_policy: Type[GroupPolicy], organization: Organization) -> NoiseConfig:
+    if group_policy.organizations and organization.slug in group_policy.organizations.keys():
+        return group_policy.organizations[organization.slug]
+
+    if organization.flags.early_adopter.is_set and group_policy.early_access:
+        return group_policy.early_access
+
+    return group_policy.default
```

Comment:
ideally this would be in the class, but typing didn't like it and kept wanting to make sure the class was instantiated